lib/network/kmsAWS/Client.ts

```diff
@@ -163,30 +135,23 @@ export default class Client implements KMSInterface {
             KeyId: masterKeyId,
             PendingWindowInDays: 7,
         };
-        this.client.scheduleKeyDeletion(params, (err: AWSError, data) => {
-            if (err) {
-                if (err.code === 'NotFoundException' || err.code === 'KMSInvalidStateException') {
-                    // master key does not exist or is already pending deletion
-                    logger.info('AWS KMS: key does not exist or is already pending deletion',
-                        { masterKeyId, error: err });
-                    cb(null);
-                    return;
-                }
-
-                const error = arsenalErrorAWSKMS(err);
-                logger.error('AWS KMS: failed to delete master encryption key', { err });
-                cb(error);
-                return;
-            }
-
+        this.client.send(new ScheduleKeyDeletionCommand(params)).then(data => {
```

Comment:
nit: I think it's nicer to always have client.send(command) and define command earlier, instead of having params as an untyped struct
```suggestion
        const command = new ScheduleKeyDeletionCommand({
            ....
        })
        this.client.send(command).then(data => {
```

Reply:
Yeah actually just like you did in the healthcheck() function at the end of this file